hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -696,3 +696,80 @@ def valid_tuple_axes(ndim, min_size=0, max_size=None):
         lambda x: x if x < ndim else x - 2 * ndim
     )
     return st.lists(axes, min_size, max_size, unique_by=lambda x: x % ndim).map(tuple)
+
+
+def _broadcastable_shapes(shape, min_dims=0, max_dims=3, min_side=0, max_side=5):
```

Comment:
This would probably be a lot easier to read as an `@composite` strategy, rather than a function returning strategy objects.

Reply:
I felt pretty crazy writing this - I was under the impression that composite was to be avoided for 1st party strategies. The refactored strategy is indeed much easier to reason about! 😅